src/buildercore/core.py

```diff
@@ -182,6 +182,10 @@ def find_ec2_instances(stackname, state='running', node_ids=None, allow_empty=Fa
         ec2_instances = [i for i in all_ec2_instances if i.state == state]
     else:
         ec2_instances = all_ec2_instances
+
+    # multiple instances are sorted by node asc
+    ec2_instances = sorted(ec2_instances, key=lambda ec2inst: ec2inst.tags.get('Node', 0))
```

Comment:
this tag should be present everywhere by now, as we updated all tags for cost tracking